ext/re2/extconf.rb

```diff
@@ -9,370 +11,326 @@
-end
+            CFLAGS
+                If this string is accepted by the compiler, add it to the flags passed to the compiler
 
-def concat_flags(*args)
-  args.compact.join(" ")
-end
+            LDFLAGS
+                If this string is accepted by the linker, add it to the flags passed to the linker
 
-def do_help
-  print(RE2_HELP_MESSAGE)
-  exit!(0)
-end
-
-def darwin?
-  RbConfig::CONFIG["target_os"].include?("darwin")
-end
-
-def windows?
-  RbConfig::CONFIG["target_os"].match?(/mingw|mswin/)
-end
-
-def freebsd?
-  RbConfig::CONFIG["target_os"].include?("freebsd")
-end
-
-def target_host
-  # We use 'host' to set compiler prefix for cross-compiling. Prefer host_alias over host. And
-  # prefer i686 (what external dev tools use) to i386 (what ruby's configure.ac emits).
-  host = RbConfig::CONFIG["host_alias"].empty? ? RbConfig::CONFIG["host"] : RbConfig::CONFIG["host_alias"]
-  host.gsub(/i386/, "i686")
-end
+            LIBS
+                Add this string to the flags passed to the linker
+      TEXT
+    end
 
-def target_arch
-  RbConfig::CONFIG['arch']
-end
+    private
 
-def with_temp_dir
-  Dir.mktmpdir do |temp_dir|
-    Dir.chdir(temp_dir) do
-      yield
+    def configure_cross_compiler
+      RbConfig::CONFIG["CC"] = RbConfig::MAKEFILE_CONFIG["CC"] = ENV["CC"] if ENV["CC"]
+      RbConfig::CONFIG["CXX"] = RbConfig::MAKEFILE_CONFIG["CXX"] = ENV["CXX"] if ENV["CXX"]
     end
-  end
-end
 
-#
-#  main
-#
-do_help if arg_config('--help')
+    def build_with_system_libraries
+      header_dirs = [
+        "/usr/local/include",
+        "/opt/homebrew/include",
+        "/usr/include"
+      ]
 
-if ENV["CC"]
-  RbConfig::MAKEFILE_CONFIG["CC"] = ENV["CC"]
-  RbConfig::CONFIG["CC"] = ENV["CC"]
-end
+      lib_dirs = [
+        "/usr/local/lib",
+        "/opt/homebrew/lib",
+        "/usr/lib"
+      ]
 
-if ENV["CXX"]
-  RbConfig::MAKEFILE_CONFIG["CXX"] = ENV["CXX"]
-  RbConfig::CONFIG["CXX"] = ENV["CXX"]
-end
+      dir_config("re2", header_dirs, lib_dirs)
+    end
 
-def build_extension(static_p = false)
-  # Enable optional warnings but disable deprecated register warning for Ruby 2.6 support
-  $CFLAGS << " -Wall -Wextra -funroll-loops"
-  $CPPFLAGS << " -Wno-register"
+    def build_with_vendored_libraries
+      message "Building re2 using packaged libraries.\n"
 
-  # Pass -x c++ to force gcc to compile the test program
-  # as C++ (as it will end in .c by default).
-  compile_options = "-x c++"
+      abseil_recipe, re2_recipe = load_recipes
 
-  have_library("stdc++")
-  have_header("stdint.h")
-  have_func("rb_gc_mark_movable") # introduced in Ruby 2.7
+      process_recipe(abseil_recipe) do |recipe|
+        recipe.configure_options << '-DABSL_PROPAGATE_CXX_STD=ON'
+        # Workaround for https://github.com/abseil/abseil-cpp/issues/1510
+        recipe.configure_options << '-DCMAKE_CXX_FLAGS=-DABSL_FORCE_WAITER_MODE=4' if MiniPortile.windows?
+      end
 
-  if !static_p and !have_library("re2")
```

Comment:
_Not_ running `have_library("re2")` is important during the static build otherwise `MakeMakefile` will add `lre2` to `LIBS` to the `Makefile` increasing the risk we'll link to the wrong version.